tests/run/relative_import_leak.srctree

```diff
@@ -0,0 +1,69 @@
+PYTHON setup.py build_ext --inplace
+
+######## setup.py ########
+
+import os
+import sys
+from contextlib import contextmanager
+from Cython.Compiler.Errors import CompileError
+from Cython.Build.Dependencies import cythonize
+from distutils.core import setup
+if sys.version_info[0] < 3:
+    from StringIO import StringIO
+else:
+    from io import StringIO
+
+@contextmanager
+def assert_stderr(expected_error):
+    sys.path.append('other')
+    sys.stderr = temp_stderr = StringIO()
+    try:
+        yield
+    except CompileError:
+        assert expected_error in temp_stderr.getvalue(), '"%s" not found in stderr' % expected_error
+    else:
+        assert False, 'The module was imported even when it should not be importable'
+    finally:
+        temp_stderr.close()
+        sys.stderr = sys.__stderr__
+
+with assert_stderr('pkg' + os.sep + "aa.pyx:1:0: 'pkg" + os.sep + "bb.pxd' not found"):
```

Comment:
`os.path.join` might be clearer? Possibly moving the two names out into variables?